### PR TITLE
feat: support of KMS 5.0

### PR DESCRIPTION
**Pull request by Manuthor**

### CODE REVIEW

File: src/kms.ts [hunk 1]
```diff
@@ -338,17 +338,29 @@ export class KmsClient {
    * @param {Uint8Array} derBytes the DER certificate/private key as bytes
```
Comment (ccorsin) [inline src/kms.ts:338]: Edit the Import phrase just above - remove 'or private key'

File: src/kms.ts [hunk 2]
```diff
@@ -369,6 +381,67 @@ export class KmsClient {
     )
   }
 
+  /**
+   * Import a X509 certificate or a private key (both as DER encoded)
```
Comment (ccorsin) [inline src/kms.ts:385]: Import a private key - only ?

File: src/kms.ts [hunk 3]
```diff
@@ -369,6 +381,67 @@ export class KmsClient {
...
+    ) {
+      attributes.link = [
+        new Link(LinkType.CertificateLink, options.certificateIdentifier),
+      ]
+    }
+    attributes.cryptographicLength = derBytes.length * 8
+
+    const privateKey = new PrivateKey(
+      new KeyBlock(
+        options.keyFormatType ?? KeyFormatType.ECPrivateKey,
```
Comment (ccorsin) [inline src/kms.ts:419]: What happens if keyFormatType is a kind of public key ? No error should be raised? We will use 'importPrivateKey' to store public keys

File: src/kms.ts [hunk 4]
```diff
@@ -369,6 +381,67 @@ export class KmsClient {
...
 
+  /**
+   * Import a X509 certificate or a private key (both as DER encoded)
+   * @param {string} uniqueIdentifier  the unique identifier of the key
+   * @param {Uint8Array} derBytes the DER certificate/private key as bytes
+   * @param {string[]} tags potential list of tags
+   * @param {boolean} replaceExisting replace the existing object
+   * @param options Additional optional options
+   * @param {string} options.keyFormatType the key format type as specified in `object_data_structures.ts`
+   * @param {string} options.certificateIdentifier the private key id in order to create the link with the private key
```
Comment (ccorsin) [inline src/kms.ts:392]: wrong phrase - certificate key id to create link with the private key

File: src/kms.ts (hunk 1 above)
Comment (Manuthor) [reply src/kms.ts]: done in a4855b7d4b9ca2562caa4bf2ac21b1153dd984cf

File: src/kms.ts (hunk 4 above)
Comment (Manuthor) [reply src/kms.ts]: thanks, don in a4855b7d4b9ca2562caa4bf2ac21b1153dd984cf

File: src/kms.ts (hunk 3 above)
Comment (Manuthor) [reply src/kms.ts:419]: actually we won't be using the public key directly but only the X509 certificate

File: src/kms.ts (hunk 2 above)
Comment (Manuthor) [reply src/kms.ts]: yes, only a private key here

**Pull request merged**